server/plugins/schema-plugin-ob-oracle/src/main/java/com/oceanbase/odc/plugin/schema/oboracle/parser/OBOracleGetDBTableByParser.java

```diff
@@ -276,44 +282,64 @@ public List<DBTableIndex> listIndexes() {
         for (DBTableIndex idx : this.indexes) {
             // ob oracle only support btree algorithm.
             idx.setAlgorithm(DBIndexAlgorithm.BTREE);
-            if (primaryKey != null && primaryKey.getName().equals(idx.getName())) {
+            String getIndexDDLSql = "SELECT dbms_metadata.get_ddl('INDEX', '" + idx.getName() + "', '" + schemaName
+                    + "') as DDL from dual";
+            // Get index type、global/local attribute and column names by parse index ddl.
+            try {
+                JdbcOperationsUtil.getJdbcOperations(connection).query(getIndexDDLSql, (rs) -> {
+                    String ddl = rs.getString("DDL");
+                    /**
+                     * Obtaining the DDL of the primary key through dbms_metadata.get_ddl will return the DDL of the
+                     * table, which is a kernel bug.
+                     */
+                    if (com.oceanbase.tools.dbbrowser.util.StringUtils.isNotBlank(ddl)
+                            && ddl.startsWith("CREATE TABLE")) {
+                        idx.setType(DBIndexType.UNIQUE);
+                        idx.setPrimary(true);
+                        idx.setUnique(true);
+                        idx.setGlobal(true);
+                        idx.setColumnNames(
+                                primaryKey.getColumnNames().stream().map(item -> removeIdentifiers(item)).collect(
+                                        Collectors.toList()));
+                    } else {
+                        CreateIndex createIndexStmt = parseIndexDDL(ddl);
+                        idx.setGlobal(createIndexStmt.getIndexOptions().getGlobal());
+                        if (createIndexStmt.getColumns().get(0).getColumn() instanceof RelationReference
+                                || createIndexStmt.getColumns().get(0).getColumn() instanceof ColumnReference) {
+                            if (createIndexStmt.isUnique()) {
+                                idx.setType(DBIndexType.UNIQUE);
+                                idx.setPrimary(false);
+                                idx.setUnique(true);
+                            } else {
+                                idx.setType(DBIndexType.NORMAL);
+                                idx.setPrimary(false);
+                                idx.setUnique(false);
+                            }
+                            idx.setColumnNames(createIndexStmt.getColumns().stream()
```

Comment:
if the create index statement is as follow, how to handle it?

```SQL
create index idx_name on idx_tbl("IDX_TBL"."COL");
```

Reply:
test ob 4.2.1 does not support this syntax：
<img width="789" alt="image" src="https://github.com/oceanbase/odc/assets/140503120/f8c6f1ba-3ab2-4468-888e-5cd4dfc8ea28">